src/Components/RandomQuestionList/utils.ts

```diff
@@ -0,0 +1,30 @@
+import { Question } from "../../Apis/types";
+
+interface Props {
+    questions: Question[];
+}
+
+export const ShuffleArray = ({ questions }: Props) => {
+    let shuffledQuestions = questions;
+
+    for (let i = 0; i < shuffledQuestions.length; i += 1) {
+        let randomIndexOne = Math.floor(
+            Math.random() * shuffledQuestions.length
+        );
+        let randomIndexTwo = Math.floor(
+            Math.random() * shuffledQuestions.length
+        );
+
+        let a = shuffledQuestions[randomIndexOne];
+        let b = shuffledQuestions[i];
+        shuffledQuestions[i] = a;
+        shuffledQuestions[randomIndexOne] = b;
+
+        let c = shuffledQuestions[randomIndexTwo];
+        let d = shuffledQuestions[i];
+        shuffledQuestions[i] = c;
+        shuffledQuestions[randomIndexTwo] = d;
```

Comment:
one shuffle was probably good, but a map() with the questions would have been better to see I think with the values shuffled in there and randomly push to a new array. this could be costly in the long run and there could be a more elegant solution.